lib/parser.js

```diff
@@ -362,22 +366,25 @@ module.exports = class Parser {
     if (node === undefined) {
       node = this._t.peek();
     }
-    return this._posStack && node.location ? node.location.pos : -1;
+    return this._posStack && node.location ? node.location.start : -1;
   }
 
   getEnd(node) {
     return this._posStack && node.location ? node.location.end : -1;
   }
 
-  finish(node, pos, end) {
-    if (pos === undefined) {
-      pos = this.popPos();
-    }
-    if (end === undefined) {
-      end = this.getPos();
-    }
+  finish(node, start, end) {
     if (this._posStack) {
-      node.location = { pos, end };
+      if (start === undefined) {
+        start = this.popPos();
+      }
+      if (end === undefined) {
+        end =
+          this._t.previous === undefined
+            ? { line: 1, column: 0, offset: 0 }
```

Comment:
A small nit: I've always been partial to `character` over `column`, as `column` can have two meanings:
- The offset of the *character* within the current line.
- The visual column from the start of the line including *tab stops*.

This is actually a very important distinction with anything "markdown" related, as various markdown dialects (including Commonmark and GFM), make that express distinction as any leading Tab character is treated as 4 space characters with respect to calculating the current block indentation level: 

https://spec.commonmark.org/0.29/#tabs
> Tabs in lines are not expanded to spaces. However, in contexts where whitespace helps to define block structure, tabs behave as if they were replaced by spaces with a tab stop of 4 characters.
>
> Thus, for example, a tab can be used instead of four spaces in an indented code block. (Note, however, that internal tabs are passed through as literal tabs, not expanded to spaces.)

Reply:
I looked into it some, and among parsers written in JavaScript there appears to be universal consensus on `column` (among parsers which track that), with the meaning it has here - it's what [jsdom](https://github.com/jsdom/jsdom/blob/11be37a64315de57a02486010b725a49a69ac796/test/api/methods.js#L71), [babel](https://github.com/babel/babel/blob/f6c7bf36cec81baaba8c37e572985bb59ca334b1/packages/babel-parser/src/util/location.js#L14), [acorn](https://github.com/acornjs/acorn/blob/d3c42927398e2c0876c2ac70a3c14da7aef85736/acorn/src/locutil.js#L9), [postcss](https://github.com/postcss/postcss/blob/a05dc46879460ac90410c2b092459fe1df3f7cfa/lib/parser.es6#L18), [angular](https://github.com/angular/angular/blob/83a9159063dc9d93c51f462eb28db3f3ad83e87b/packages/compiler/src/css_parser/css_lexer.ts#L124), etc all use, and is what the [sourcemap spec](https://sourcemaps.info/spec.html) requires. I'm going to stick with the JS ecosystem convention here.